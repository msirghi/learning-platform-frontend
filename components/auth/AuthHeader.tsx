import React from 'react';
import { AuthPage } from '../../common/enums';
import { useTranslation } from '../../i18n';
import styles from '../../styles/modules/Auth.module.scss';

/**
 * AuthHeader component props
 */
type Props = {
  /**
   * Function fired on tab change.
   */
  onTabChange: (tab: AuthPage) => void;
  /**
   * Currently active tab.
   */
  activeTab: AuthPage;
};

/**
 * Header from the auth page. Includes two tabs: Registration & Login.
 *
 * @version 0.1
 * @author [Sirghi Mihail](https://github.com/msirghi)
 */
function AuthHeader({ onTabChange, activeTab }: Props) {
  const isRegisterTab = activeTab === AuthPage.REGISTER;
  const { t } = useTranslation();

  return (
    <div className={styles.authHeader}>
      <div
        className={`${!isRegisterTab && styles.inactiveTab} ${styles.tab} ${styles.registerTab}`}
        onClick={() => onTabChange(AuthPage.REGISTER)}
      >
        <span>{t('auth:registerTitle')}</span>
      </div>
      <div
        className={`${isRegisterTab && styles.inactiveTab} ${styles.tab} ${styles.loginTab}`}
        onClick={() => onTabChange(AuthPage.LOGIN)}
      >
        <span>{t('auth:loginTitle')}</span>
      </div>
    </div>
  );
}

AuthHeader.getInitialProps = async () => ({
  namespacesRequired: ['auth']
});

export default AuthHeader;
