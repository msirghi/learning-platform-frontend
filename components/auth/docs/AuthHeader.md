React component example:

```tsx
<AuthHeader size='large'>Push Me</AuthHeader>
```
Any [Markdown](http://daringfireball.net/projects/markdown/) is **allowed** _here_.
