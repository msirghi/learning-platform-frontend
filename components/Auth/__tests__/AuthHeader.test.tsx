import { shallow, ShallowWrapper } from 'enzyme';
import AuthHeader from '../AuthHeader';
import toJson from 'enzyme-to-json';

describe('AuthHeader component', () => {
  let component: ShallowWrapper;

  const defaultProps = {
    onTabChange: jest.fn(),
    activeTab: 1
  };

  beforeEach(() => {
    component = shallow(<AuthHeader {...defaultProps} />);
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should fire the prop function on register tab click', () => {
    const tab = component.find('.registerTab');
    tab.simulate('click');
    expect(defaultProps.onTabChange).toBeCalledTimes(1);
  });

  it('should fire the prop function on login tab click', () => {
    jest.clearAllMocks();
    const tab = component.find('.loginTab');
    tab.simulate('click');
    expect(defaultProps.onTabChange).toBeCalledTimes(1);
  });

  it('should have inactive register tab if login tab is active', () => {
    const registerTab = component.find('.registerTab');
    const loginTab = component.find('.loginTab');

    expect(registerTab.hasClass('inactiveTab')).toBeTruthy();
    expect(loginTab.hasClass('inactiveTab')).toBeFalsy();
  });

  it('should have inactive login tab if register tab is active', () => {
    const component = shallow(<AuthHeader onTabChange={jest.fn()} activeTab={0} />);
    const registerTab = component.find('.registerTab');
    const loginTab = component.find('.loginTab');

    expect(registerTab.hasClass('inactiveTab')).toBeFalsy();
    expect(loginTab.hasClass('inactiveTab')).toBeTruthy();
  });
});
