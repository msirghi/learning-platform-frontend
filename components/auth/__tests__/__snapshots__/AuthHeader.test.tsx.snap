// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeader component should match snapshot 1`] = `
<div
  className="authHeader"
>
  <div
    className="inactiveTab tab registerTab"
    onClick={[Function]}
  >
    <span>
      auth:registerTitle
    </span>
  </div>
  <div
    className="false tab loginTab"
    onClick={[Function]}
  >
    <span>
      auth:loginTitle
    </span>
  </div>
</div>
`;
